src/WebJobs.Script.WebHost/Middleware/HstsConfigurationMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class HstsConfigurationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IScriptWebHostEnvironment _webHostEnvironment;
+        private HstsMiddleware _hstsMiddleware;
+        private HostHstsOptions _hostHstsOptions;
+
+        public HstsConfigurationMiddleware(RequestDelegate next, IScriptWebHostEnvironment webHostEnvironment)
+        {
+            _next = next;
+            _webHostEnvironment = webHostEnvironment;
+        }
+
+        public async Task Invoke(HttpContext httpContext, IOptions<HostHstsOptions> options)
```

Comment:
If host is offline or shut down, you'll still get an IOptions<HostHstsOptions> here (I think), but it'll be a default object. Need to make sure that it's evaluating to false in that case (which it should be). Is there a test for that?

Reply:
I don't think there is a test case for it. The default value should be false, but i can explicitly set the default to false and add a test case.

Reply:
The host configuration would still be applied in this case. The case where it wouldn't is if there's an initialization failure due to a configuration parsing error.